build/v1/types.go

```diff
@@ -730,6 +730,8 @@ type DockerBuildStrategy struct {
 
 	// buildArgs contains build arguments that will be resolved in the Dockerfile.  See
 	// https://docs.docker.com/engine/reference/builder/#/arg for more details.
+	// NOTE: Only the 'name' and 'value' fields are supported. Any settings on the 'valueFrom' field
```

Comment:
Has it ever been honored before?  This sort of a comment is an indication that the schema being embedded isn't actually the schema we wanted: it was just similar at some point in time.  

If it has never been honored, then creating a type with the schema you want seems more appropriate.

Reply:
It has never been honored.  And absolutely in hindsight choosing corev1.EnvVar was not the best choice here.

Admittedly we went with this band aid since we are coming up on code freeze.  I'll ultimately defer to @adambkaplan on this, but at least for me it was too late in the cycle to take on a schema change.

WDYT @adambkaplan ?

Reply:
it seems that builds are actually serialized as an API into an env var (as I recall), which a build could then rely upon.  We have to keep this field now.

In the future, it's still ok to rely on an upstream schema in cases where you are simply passing it around blindly (deploymentconfig) for instance.  But in cases where you are interpreting the meaning of fields, the schema is very rarely actually the upstream one.